4_gap_analysis/src/3_conusMap.R

```diff
@@ -0,0 +1,75 @@
+#setwd("D:/abock/NAWQA/WQR/nawqa_wqp")
+
+# Bugs in tidyverse interfere with ggplot2 maps functionality
+# https://stackoverflow.com/questions/45066628/cannot-run-map-datastate
+# library(tidyverse)
+library(leaflet)
+library(tidyverse)
+library(mapview)
+library(htmlwidgets)
+library(webshot)
+
+#sapply(list.files("4_gap_analysis/src",full.names = T),source,.GlobalEnv)
+siteFiles<-dir('4_gap_analysis/in/Site', pattern='*sites.feather',full.names = T)
+# Files generated by filtering (by site)
+sampFiles<-dir('4_gap_analysis/in/Tiers', pattern='*.feather',full.names = T)
+
+# combined data may not work
+siteInfo <- bind_rows(
+  lapply(siteFiles, function(dfile) {
+    dat <- feather::read_feather(dfile)
+  })
+)
+siteInfo<-siteInfo %>% distinct(MonitoringLocationIdentifier,.keep_all=T)
+
+sampleInfo<- bind_rows(
+  lapply(sampFiles, function(dfile) {
+    dat <- feather::read_feather(dfile)
+  })
+)
+
+TierA<-sampleInfo %>%
+  filter(numYears > 4 & numCon==3 & Quart > 19) %>%
+  mutate(Tier="a) 3 consituents, 5 or more years, all quarters")
+
+TierB<-sampleInfo %>%
+  filter(numYears > 4 & numCon==3 & Quart < 20)%>%
+  mutate(Tier="b) 3 cons., less than quarterly")
+
+TierC<-sampleInfo %>%
+  filter(numYears > 4 & numCon<3 & Quart > 19)%>%
+  mutate(Tier="c) 2 constituents, 5 or more years, all quarters")
+
+Tiers<-do.call("rbind",list(TierA,TierB,TierC))
+
+siteSamp<-dplyr::inner_join(x=siteInfo,y=Tiers,by="MonitoringLocationIdentifier")
+siteSamp<-siteSamp %>% filter(dec_lon_va != 0 | dec_lat_va !=0) %>% filter(!is.na(NumMnths))
+
+# Needs to be a factor
+siteSamp$TierFac<-factor(siteSamp$Tier)
+
+# Make a simple map
+m=leaflet()
+m=addTiles(m)
+colorFactors = colorFactor(c('blue','green','red'),
+                           domain = siteSamp$TierFac) #,
+
+m = addCircleMarkers(m,
+                     lng = siteSamp$dec_lon_va, # we feed the longitude coordinates
+                     lat = siteSamp$dec_lat_va,
+                     popup = siteSamp$Tier,#OrganizationIdentifier,
+                     radius = 3,
+                     stroke = FALSE,
+                     fillOpacity = 0.75,
+                     col=colorFactors(siteSamp$TierFac),
+                     group = "1 - graffiti & noise"
+) %>% addLegend(title = "Nitrate, dissolved phosphorus, and total phosphorus <br> Monthly sampling at WQP sites, 2010-2015",
+                  "bottomright",pal=colorFactors,values=siteSamp$TierFac) %>%
+  setView(-96.5795,39.8283,zoom=3.3)
```

Comment:
Are there sites in AK, HI, PR, etc. that would be useful to include in this map?

Reply:
For some reason, I was under the impression we were just looking at CONUS, but I'll double-check with Lori.